### DISCUSSION TITLE
Allow customizable default of RPMBUILD_RMBUILD in rpmbuild (--noclean)

### DISCUSSION THREAD

**Discussion by pemensik** [rpm-software-management/rpm#2942] [Ideas]
**Is your feature request related to a problem? Please describe.**
Before commit b34333fa021c0ee7215714eeef96d1a2843ea08e, rpmbuild has by default kept built artifacts. I had often used it to run upstream test suite of bind package, which needs first setup made by root. Therefore it could not be done during normal build. Also it takes long.

When something fails there, I may need use gdb or other debug steps to find a solution. It helps me a lot if the gdb can reuse my prepared settings, not vanilla mock environment.

**Describe the solution you'd like**
I would like to be able to change default of --clean for rpmbuild -bb or -ba. I want to override it on my machine, so it by default keeps built artifacts for subsequent testing or debugging. I would like to avoid the need to append --noclean to every build. Instead some kind of configuration change to some file, done only once on my system.

**Describe alternatives you've considered**
Making alias, which would add rpmbuild --noclean parameter to everything might work. But such alias would not be used, if I use rpkg variants such as fedpkg to build my package. I would have to make aliases for that also. But there it will become complicated, rpmbuild options to it need to be last arguments. I would have to make special shell wrapper. That makes it unnecessary complicated. Just default change would work much better.

**Additional context**
I haven't found a way to configure the default. It does not seem to be possible to change default value in ~/.rpmmacros or any configuration file. Deleting the results is much faster than recreating it every time I forgot --noclean appended.

Opinions on this topic? Please skip the phase use mock instead. It does not work the best for me.

**Comment by pemensik** [comment]
In fact, I might want to delete built rpms instead. Usually I do not need them anyway, because I do local builds usually just o test something with built version. I need builddir, but not locally generated rpms. But that would be other topic.

**Comment by Conan-Kudo** [comment]
> Before commit https://github.com/rpm-software-management/rpm/commit/b34333fa021c0ee7215714eeef96d1a2843ea08e, rpmbuild has by default kept built artifacts.

~~RPM has deleted the buildroot tree by default since RPM 4.6 in nearly all distributions. If it wasn't doing that before, that's a bug.~~

Oh wait, this is if clean is not executed. I don't think a customizable default makes a lot of sense, but maybe a flag to avoid it?

**Reply by pemensik** [reply]
Tags on mentioned commit say:
Follows: rpm-4.17.0-alpha
Precedes: rpm-4.19.0-alpha
Just tried it on rpm-build-4.16.1.3-29.el9.s390x, where --noclean is not necessary. This is not about deleting buildroot *before* the build, but *after successful* build.

**Comment by pmatilai** [comment]
If you don't care about binaries then why are you building them in the first place?
'rpmbuild -bi' will do all the same build steps, only it wont procuce packages or clean the build, because it's *intended* for that sort of work.

**Reply by pemensik** [reply]
I want to check unit tests are passing in my builds. Also I want to check %files are found and packaged, but most of time I will delete built rpms without installing them. Afaik *fedpkg* does not allow to use rpmbuild -bi alternative, but has -bb via local build :)

**Reply by pemensik** [reply]
seems ``rpmbuild -bl`` would be ideal for me.

**Reply by pmatilai** [reply]
rpmbuild -bi does run %check, and it also runs the usual files checks. Like I said, it basically does everything but produce binaries.

`fedpkg install` (it's a weird name for what it does, yes) is the fedpkg equivalent.

**Reply by pemensik** [reply]
Ah, then yes, that would be what I were looking for. But because quite unintuitive name I have never tried to use it.
